modules/common/util/template_util.go

```diff
@@ -135,6 +176,12 @@ func ExecuteTemplate(templateFile string, data interface{}) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
```

Comment:
ExecuteTemplateData is called from different places to render templatest:
* here from ExecuteTemplate
* but also from secret.createOrUpdateSecret()

Moreover the actual template rendering happens via the text/template module by creating a template via template.New() and eventually calling Execute() on it. If you search for that pattern in this file then you will see that there are two functions that actually executes templates:
* ExecuteTemplateData 
* ExecuteTemplateFile

So both call path need to be handled. The nova.conf rendering goes through the ExecuteTemplateFile path this is why you don't see your cleaning applied to that config file.

Reply:
I would do the following:
* pull the template rendering part of the two paths to a common function
* pass down the template_utils.Template struct to that common function
* extend the template_utils.Template struct with a field that can configure if / what post processing is needed for the give template
* use that information in the common function to do the post processing.

Reply:
Added bool in Template to make cleanup optional